src/components/ExportCapture.js

```diff
@@ -0,0 +1,122 @@
+    flexGrow: 1,
+  },
+  formControl: {
+    margin: theme.spacing(3),
+  },
+  csvLink: {
+    textDecoration: "none",
+    color: theme.palette.stats.green,
+  }
+}));
+
+const ExportCapture = (props) => {
+  const { isOpen, handleClose, columns, filter, speciesState } = props;
+  const classes = useStyle();
+  let nameColumns = {};
+  columns.forEach(({ attr }) => {
+    nameColumns[attr] = true;
+  });
+  const [checkedColumns, setCheckColumns] = useState(nameColumns);
+  const capturesContext = useContext(CapturesContext);
+  const [downloadData, setDownloadData] = useState([]);
+
+  function handleChange(attr) {
+    setCheckColumns({ ...checkedColumns, [attr]: false });
+  }
+
+  function processDownloadData(captures, selectedColumns) {
+    return captures.map(capture => {
+      let formatCapture = {};
+      Object.keys(selectedColumns).forEach(attr => {
+        if (attr === 'id' || attr === 'planterId') {
+          formatCapture[attr] = capture[attr];
+        } else if (attr === 'tokenId') {
+          formatCapture[attr] = formatCell(capture, speciesState, attr, (val) =>
+          val ? tokenizationStates.TOKENIZED : tokenizationStates.NOT_TOKENIZED);
+        } else if (attr === 'timeCreated') {
+          formatCapture[attr] = formatCell(capture, speciesState, attr, (val) => getDateTimeStringLocale(val));
+        } else {
+          formatCapture[attr] = formatCell(capture, speciesState, attr, null);
+        }
+      })
+      return formatCapture;
+    })
+  }
+
+  return (
+    <Dialog
+      open={isOpen}
+      onClose={handleClose}
+      aria-labelledby="form-dialog-title"
+    >
+      <DialogTitle id="form-dialog-title">Export Captures</DialogTitle>
+      <DialogContent>
+        <FormControl component="fieldset" className={classes.formControl}>
+          <FormGroup>
+            {columns.map(({ attr, label }) => (
+              <FormControlLabel
+                key={attr}
+                control={
+                  <Checkbox
+                    checked={checkedColumns[attr]}
+                    onChange={() => handleChange(attr)}
+                    name={attr}
+                  />
+                }
+                label={label}
+              />
+            ))}
+          </FormGroup>
+        </FormControl>
+      </DialogContent>
+      <DialogActions>
+        <Button onClick={handleClose}>Cancel</Button>
+        <Button color="primary">
+          <CSVLink
+            className={classes.csvLink}
+            data={downloadData}
+            filename={"captures.csv"}
+            asyncOnClick={true}
+            onClick={() => {
+              let filterColumns = Object.entries(checkedColumns).filter((val) => val[1] === true);
+              const selectedColumns = Object.fromEntries(filterColumns);
+              capturesContext.getAllCaptures({ filter }).then((response) => {
+                setDownloadData(processDownloadData(response.data, selectedColumns));
+              });
```

Comment:
The _Export Captures_ dialog should close when the user clicks `Download CSV`, and be replaced by the visual feedback on progress mentioned in the comment above.